cypress/utils/dashboards/workspace-plugin/commands.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BASE_PATH } from '../../base_constants';
+import { WORKSPACE_API_PREFIX } from './constants';
+
+Cypress.Commands.add('deleteWorkspace', (workspaceName) => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_PATH}${WORKSPACE_API_PREFIX}/_list`,
+    headers: {
+      'osd-xsrf': true,
+    },
+    body: {},
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      resp.body.result.workspaces.map(({ name, id }) => {
+        if (workspaceName === name) {
+          cy.request({
+            method: 'DELETE',
+            url: `${BASE_PATH}${WORKSPACE_API_PREFIX}/${id}`,
+            headers: {
+              'osd-xsrf': true,
+            },
+          });
+        }
+      });
+    }
+  });
+});
+
+Cypress.Commands.add('createWorkspace', (workspace) => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_PATH}${WORKSPACE_API_PREFIX}`,
+    headers: {
+      'osd-xsrf': true,
+    },
+    body: {
+      attributes: {
+        name: workspace.name,
+        description: 'test_description',
+      },
+    },
+  }).then((resp) => {
+    if (resp && resp.body && resp.body.success) {
+      workspace.id = resp.body.result.id;
+    } else {
+      throw new Error(`Create workspace ${workspace} failed!`);
+    }
+  });
+});
+
+Cypress.Commands.add('checkWorkspace', (workspaceName, expected) => {
```

Comment:
Shall we checkWorkspace based the workspace id? So we don't need to call list API and use workspace name to match worksapces.